version: '3'
services:
  nginx-certbot:
    build: .
    container_name: nginx-certbot
    env_file: .env
    image: rabbotio/nginx-certbot
    restart: on-failure
    volumes:
      - etc-nginx-conf.d:/etc/nginx/conf.d
      - etc-ssl:/etc/ssl
      - var-www-html:/var/www/html
      - etc-letsencrypt:/etc/letsencrypt
      - var-log-letsencrypt:/var/log/letsencrypt
      - ./root:/root
    links:
      - nginx
      - certbot
    command:
      - "init.sh"

  nginx:
    image: nginx:alpine
    container_name: nginx
    env_file: .env
    restart: on-failure
    networks:
      - back
    volumes:
      - etc-nginx-conf.d:/etc/nginx/conf.d
      - etc-ssl:/etc/ssl
      - var-www-html:/var/www/html
    logging:
      driver: json-file
      options:
        max-size: "10m"
        max-file: "5"
    ports:
      - "80:80"
      - "443:443"

  certbot:
    image: certbot/certbot
    container_name: certbot
    env_file: .env
    restart: on-failure
    environment:
      - DOMAIN=$DOMAIN
      - CERTBOT_EMAIL=$CERTBOT_EMAIL
      - ACME_WWWROOT=${ACME_WWWROOT:-/var/www/html}
    networks:
      - back
    volumes:
      - etc-letsencrypt:/etc/letsencrypt
      - var-log-letsencrypt:/var/log/letsencrypt

volumes:
  etc-nginx-conf.d:
  etc-ssl:
  var-www-html:
  etc-letsencrypt:
  var-log-letsencrypt:

networks:
  back:
